### PR TITLE
Javascript bundles optimization 

### PR DESCRIPTION
**Pull request by Oksydan**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Vendors are split into two vendors chunks: graphql and other apollo, vue. Bundles file size has been reduced. Bundle analyzer has been added for prod build.
| Type?             | improvement
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/24255.
| How to test?      | Try tu run `npm run watch` or `npm run build` and check if bundles are loading correctly. The module should works as expected. If you want to test further, you can also check in network tabs the size of bundles before and after this PR :)

@NeOMakinG

### CODE REVIEW

**Comment by kpodemski** [comment]
Hey @Oksydan 

Thank you for your contribution 🔥 

could you give us quick comparison `before` and `after` these changes?

**Comment by Oksydan** [comment]
@kpodemski Yes of course. Around 80% size reduction. 

### Before 
before

### After
after

File: .webpack/prod.js [hunk 1]
```diff
@@ -23,16 +27,14 @@ function prodConfig() {
...
-}
+    plugins: [
+      new webpack.DefinePlugin({
+        'process.env.NODE_ENV': JSON.stringify('production'),
+      }),
+      new BundleAnalyzerPlugin(),
+    ],
+  },
+)
+);
```
Comment (NeOMakinG) [inline .webpack/prod.js:38]: Hey @Oksydan 

I've a little question about this, do we really need webpack-merge? A spread operator will do the same thing don't you think?

**Comment by NeOMakinG** [comment]
Thanks for this contribution, this looks really good to me, only one question left :)

File: .webpack/prod.js (hunk 1 above)
Comment (NeOMakinG) [reply .webpack/prod.js:38]: Got my response through webpack-merge doc, it's different

**Review by NeOMakinG — APPROVED** [review]
🙏

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

File: .webpack/prod.js (hunk 1 above)
Comment (Oksydan) [reply .webpack/prod.js:38]: @NeOMakinG 
I recommend using webpack-merge for merging configs. I will save you some confusion caused by deep merge problem.

Comment (NeOMakinG) [reply .webpack/prod.js:38]: It's going deeper than even the deepest merge you can do using JS, seen it ;)

**Comment by Hlavtox** [comment]
Guys this needs to be **really** optimized. I just tried to install the current version of the module and my website javascript bundle went from **86 kB** to **246 kB**???

**Comment by Oksydan** [comment]
@Hlavtox 
Currently there is no way to go lower w/o making a lot of changes.
Just technology stack that was selected for this module contains big libraries.
There is possibility to go lower with dynamic import for some libraries but it will require a lot of work and I am not sure if it is possible w/o removing graphql completely. 
I am not that into module structure  and I am not sure if it is making first request on page load thru graphql.
If it does lazy loading it will only make request like 1s later, so there is no point of doing it.
@NeOMakinG correct me if I am wrong.

**Comment by Oksydan** [comment]
@NeOMakinG 
I did quick check and removing graphql and apollo will reduce size by 90 kB. Vendor chunk contains apollo and vue-apollo libraries. 
Btw we can provide fetch polyfill or just use jQuery ajax method.

**Comment by Hlavtox** [comment]
@Oksydan @NeOMakinG After Oksydan's fix it's much better! :-)

I initially thought we could load the full bundle only on my-account pages and keep only "adding" functionality for the rest of the pages, loading only single kBs of resources, but I see I can do everything from everywhere.

**Comment by NeOMakinG** [comment]
I removed the bundle analyzer, the CI was broken by it

**Review by kpodemski — APPROVED** [review]

**Comment by sarahdib** [comment]
Hello @Oksydan 

Thanks for this contribution.

Could you fix the conflicting files please ? 

Thank you :)

**Comment by Oksydan** [comment]
@sarahdib done

I think we should consider include `.nvmrc` to every webpack config modules/themes(BO, FO). It took me some time to grab right node version to make sure that assets are build correctly. @NeOMakinG what you think?

**Comment by kpodemski** [comment]
Hey @Oksydan 

Could you rebase your PR once again? We'll try to find a way to stop versioning bundled files to avoid this issue in the future, for now, please fix conflicts once again so QA can validate your PR 👍 

Much thanks!

**Comment by Oksydan** [comment]
@kpodemski done

**Comment by kpodemski** [comment]
Thank you @Oksydan @sarahdib

**Pull request merged**